server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -816,6 +830,12 @@ private Function optimizedSubscriptFunction(Symbol base,
                 if (e != null) {
                     throw e;
                 }
+                ColumnUnknownException.analysisSubscriptError(base, path.getLast());
```

Comment:
This is a bit odd that `analysisSubscriptError` is void.

Could we have something like `throw ColumnUknownException.forSubscript(base, path)` here, and move the fallback also into the `forSubscript` method?